### ISSUE TITLE
install_subdir should keep

### ISSUE THREAD

**Issue by taozuhong** [mesonbuild/meson#11200]
`install_subdir` for whole sub directory, maybe the empty directory, it should be kept in future versions

**Comment by eli-schwartz** [comment]
I'm not sure I understand the question. Why wouldn't this function be kept in future versions of Meson?

**Comment by taozuhong** [comment]
What is the replacement of `install_subdir`? `install_emptydir` can't solve all problems.

**Comment by eli-schwartz** [comment]
It's not supposed to, though. It's only supposed to solve creating an empty directory.

`install_subdir` isn't deprecated. The only thing that is deprecated is manipulating a Meson bug to use `install_subdir` to install a directory that doesn't exist.

**Comment by taozuhong** [comment]
Maybe `install_directory` is the choice for copy whole directory with subdir and files, even if the folder is not exist(create empty directory)。

**Comment by eli-schwartz** [comment]
No, that's just the original intended purpose of `install_subdir()`, which is still the official way to do it.

> deprecated since 0.60.0) If the subdirectory does not exist in the source tree, an empty directory is created in the specified location.

That is the only thing that is deprecated.

```
void install_subdir(
    str subdir_name,     # The sub-directory to install

    install_dir         : str              # Where to install to
)
```

subdir_name must exist. If it doesn't exist, it is *deprecated* to "just create an empty directory in the install_dir". It is not the intended purpose of install_subdir().

**Comment by taozuhong** [comment]
I think should deprecate the `install_emptydir`, it's wired, or rename to `create_directory`, and do it recursively.

**Comment by eli-schwartz** [comment]
- I'm not sure I understand what you mean w.r.t. how it is internally wired up?
- It is already recursive. Leading directories are created as well.
- All install-only functions are named `install_*`, and I don't see a benefit to just renaming this. For example, `create_directory` would imply it can be used to create directories in the source tree, which is not true. `install_emptydir` makes it clear that it can only be created at install time in the destination tree.

**Comment by taozuhong** [comment]
> I'm not sure I understand what you mean w.r.t. how it is internally wired up?
It can be done with `install_subdir`, exist or not exist, its OK for all scenes. 

if you want to keep `install_emptydir`, then `install_subdir` just dropping empty directories and do not throw exceptions.

> makes it clear that it can only be created at install time in the destination tree.
I do agree with you.

**Comment by taozuhong** [comment]
> I'm not sure I understand what you mean w.r.t. how it is internally wired up?
It can be done with `install_subdir`, exist or not exist, it's OK for all scenes. if you want to keep `install_emptydir`, then `install_subdir` just dropping empty and do not throw exception.

> makes it clear that it can only be created at install time in the destination tree.
I do agree with you.
